### PR TITLE
CW-705: implement usedAddress in a more performant way

### PR DESCRIPTION
**Pull request by MrCyjaneK**
# Description

Fixes the issue that @tuxpizza reported with large wallets

# Pull Request - Checklist  

- [ ] Initial Manual Tests Passed
- [ ] Double check modified code and verify it with the feature/task requirements
- [ ] Format code
- [ ] Look for code duplication
- [ ] Clear naming for variables and methods

### CODE REVIEW

File: cw_monero/lib/api/transaction_history.dart [hunk 1]
```diff
@@ -16,20 +18,59 @@ String getTxKey(String txId) {
...
+    monero.debugCallLength["CW_Transaction_list_construct_cached"]!.add(stopwatch?.elapsedMicroseconds??0);
+    txHistoryMutex.release();
+  } else {
+    final List<Transaction> newList = [];
+    for (var i = 0; i < size; i++) {
+      newList.add(
+        Transaction(txInfo: monero.TransactionHistory_transaction(txhistory!, index: i))
+      );
+      if (enableDelay) {
+        await Future.delayed(Duration.zero);
```
Comment (konstantinullrich) [inline cw_monero/lib/api/transaction_history.dart:61]: What is the point of this Future?

File: cw_monero/lib/monero_subaddress_list.dart [hunk 2]
```diff
@@ -143,8 +147,7 @@ abstract class MoneroSubaddressListBase with Store {
                     label.toLowerCase() == 'Primary account'.toLowerCase()
                 ? 'Primary address'
                 : label);
-      })
-        .toList();
+      }).toList().reversed.toList();
```
Comment (konstantinullrich) [inline cw_monero/lib/monero_subaddress_list.dart:150]: Why reverse it here?

File: cw_monero/lib/api/transaction_history.dart (hunk 1 above)
Comment (MrCyjaneK) [reply cw_monero/lib/api/transaction_history.dart:61]: to give the UI some time to render, future is being put at the end of the scheduler queue (reusing my JS knowledge here, but it works)

File: cw_monero/lib/monero_subaddress_list.dart (hunk 2 above)
Comment (MrCyjaneK) [reply cw_monero/lib/monero_subaddress_list.dart:150]: otherwise addresses appear upside down in the receive screen

**Pull request closed without merging**